pact/Pact/Core/TransitiveDependencies.hs

```diff
@@ -111,36 +170,85 @@ canHaveDependents = \case
   DPact{} -> True
   _ -> False
 
--- | Get all transitive depencen
+popWorkingSet :: TransitiveClosureM b i (Maybe FullyQualifiedName)
+popWorkingSet =
+  uses workingSet popFromWorkingSet >>= \case
+    Nothing -> pure Nothing
+    Just (e, rest) -> do
+      workingSet .= rest
+      pure (Just e)
+
+lookupFqn :: FullyQualifiedName -> TransitiveClosureM b i (EvalDef b i)
+lookupFqn fqn = uses allDependencies (M.lookup fqn) >>= \case
+  Just defn -> pure defn
+  Nothing -> throwError (TCUnboundFreeVarError fqn)
+
+-- | Compute all transitive dependencies from our initial working set.
+--   The algorithm is pretty simple: If a particular component can have dependents,
+--   then we traverse its tree and collect all of its dependents, and push those not currently in
+--   the dependency set into it as well as the working set.
+--
+--   The algorithm only ever grows the dependency set, it will never shrink it, thus
+--   it is guaranteed to terminate
+computeTransitiveDeps :: TransitiveClosureM b i ()
+computeTransitiveDeps = computeUntilDone
```

Comment:
This is my favorite part of this PR. By floating the algorithm out of `getAllTransitiveDependencies`, it simplified _drastically_